packages/fiber/src/core/reconciler.tsx

```diff
@@ -309,7 +317,12 @@ function switchInstance(
   if (oldInstance.isHidden) unhideInstance(oldInstance)
 
   // Create a new instance
-  const newInstance = createInstance(type, props, oldInstance.root)
+  const newInstance = createInstance(type, props, oldInstance.root, false)
+
+  // Update attach props for primitives since we don't flush them
+  if (type === 'primitive') {
+    newInstance.props.attach = props.attach
+  }
```

Comment:
Probably an issue with `applyProps` not being called for primitives in `prepare`.